metadata/dashboard/dashboard_metadata.go

```diff
@@ -12,32 +12,79 @@ import (
 	"strings"
 
 	help "github.com/featureform/helpers"
+	"github.com/featureform/metadata"
+	pb "github.com/featureform/metadata/proto"
 	"github.com/featureform/metadata/search"
 	"github.com/featureform/proto"
 	"github.com/featureform/provider"
 	pt "github.com/featureform/provider/provider_type"
 	"github.com/featureform/serving"
-	"github.com/pkg/errors"
-
-	"github.com/featureform/metadata"
 	"github.com/gin-contrib/cors"
 	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"golang.org/x/sync/errgroup"
 )
 
 var searchClient search.Searcher
 
+type StorageProvider interface {
+	GetResourceLookup() (metadata.ResourceLookup, error)
+}
+
+type LocalStorageProvider struct {
+}
+
+func (sp LocalStorageProvider) GetResourceLookup() (metadata.ResourceLookup, error) {
+	lookup := make(metadata.LocalResourceLookup)
+	return lookup, nil
+}
+
+type EtcdStorageProvider struct {
+	Config metadata.EtcdConfig
+}
+
+func (sp EtcdStorageProvider) GetResourceLookup() (metadata.ResourceLookup, error) {
+
+	client, err := sp.Config.InitClient()
+	if err != nil {
+		return nil, fmt.Errorf("could not init etcd client: %v", err)
+	}
+	lookup := metadata.EtcdResourceLookup{
+		Connection: metadata.EtcdStorage{
+			Client: client,
+		},
+	}
+	return lookup, nil
+}
+
 type MetadataServer struct {
-	client *metadata.Client
-	logger *zap.SugaredLogger
+	lookup          metadata.ResourceLookup
+	client          *metadata.Client
+	logger          *zap.SugaredLogger
+	StorageProvider StorageProvider
 }
 
 func NewMetadataServer(logger *zap.SugaredLogger, client *metadata.Client) (*MetadataServer, error) {
 	logger.Debug("Creating new metadata server")
+	etcdHost := help.GetEnv("ETCD_HOST", "featureform-etcd")
```

Comment:
Typically we want to pass configs from a top level main function to this function (add a storage provider param), so we dont have env variables in the function itself

Reply:
gotcha, reworking